clang/lib/CodeGen/CoverageMappingGen.cpp

```diff
@@ -339,8 +365,18 @@ class CoverageMappingBuilder {
 
     llvm::SmallSet<FileID, 8> Visited;
     SmallVector<std::pair<SourceLocation, unsigned>, 8> FileLocs;
-    for (const auto &Region : SourceRegions) {
+    for (auto &Region : SourceRegions) {
       SourceLocation Loc = Region.getBeginLoc();
+
+      // Replace Region with its definition if it is in <scratch space>.
```

Comment:
Now this collides to #91446 . Could you update?
I suggest this may come in advance of #91446.